app/views.py

```diff
@@ -28,23 +28,53 @@ def home():
     """Renders the home page."""
     if current_user.is_authenticated:
         return redirect(url_for('dashboard'))
-    form = LoginForm()
-    if form.validate_on_submit():
-        user = User.query.filter_by(email=form.email.data).first()
-        if user is None or not user.check_password(form.password.data):
+    loginForm = LoginForm(prefix='login')
+    codeForm = CodeForm(prefix='code')
+    if loginForm.validate_on_submit():
+        user = User.query.filter_by(email=loginForm.email.data).first()
+        if user is None or not user.check_password(loginForm.password.data):
             flash('Invalid username or password')
             return redirect(url_for('login'))
         login_user(user)
         return redirect(url_for('dashboard'))
-    return render_template('index.html', form=form)
+    return render_template('index.html', loginForm=loginForm, codeForm=codeForm)
+
+@app.route('/progressreport/<int:knewbie_id>', methods=['GET','POST'])
+def get_report():
+    """Renders the class page for students/educators."""
+    loginForm = CreateName(prefix='login')
+    codeForm = CreateName(prefix='code')
```

Comment:
```suggestion
    loginForm = LoginForm(prefix='login')
    codeForm = CodeForm(prefix='code')
```
`LoginForm` and `CodeForm` shouldn't change from `home`. Since they are different forms I believe the `prefix` doesn't matter but just leave them in case.